src/freenet/clients/http/wizardsteps/SECURITY_PHYSICAL.java

```diff
@@ -182,8 +195,10 @@ private boolean errorHandler(HTTPRequest request, PageHelper helper) {
 	public String postStep(HTTPRequest request) throws IOException {
 		final String errorCorrupt = FirstTimeWizardToadlet.WIZARD_STEP.SECURITY_PHYSICAL+"&error=corrupt";
 		String pass = request.getPartAsStringFailsafe("masterPassword", SecurityLevelsToadlet.MAX_PASSWORD_LENGTH);
-		final boolean passwordIsBlank = pass != null && pass.length() == 0;
-
+		String confirmPass = request.getPartAsStringFailsafe("confirmMasterPassword", SecurityLevelsToadlet.MAX_PASSWORD_LENGTH);
+		final boolean passwordIsBlank = pass != null && pass.length() == 0 && confirmPass != null && confirmPass.length() == 0;
+		final boolean passwordsDoNotMatch = !pass.equals(confirmPass);
```

Comment:
The `pass != null` check in the line above implies `pass` can be null, but it can't be. The existing code acted like it can, but it can't. (We could clean that up in existing code in another pull request.) [`getPartAsStringFailsafe`](https://github.com/freenet/fred/blob/c9bcbb88dc28034e4dafda583958f8612b89c0ac/src/freenet/clients/http/HTTPRequestImpl.java#L683) prevents it - even in the case of an encoding problem in [`getPartAsLimitedString`](https://github.com/freenet/fred/blob/c9bcbb88dc28034e4dafda583958f8612b89c0ac/src/freenet/clients/http/HTTPRequestImpl.java#L691) where the behavior of the String constructor [is undefined](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#String%28byte[],%20java.lang.String%29), my understanding of this part of [the Java specification](https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.9.4) is that new instance creation cannot return null, so there is no need to check return values of `getPartAsStringFailsafe` against null - at most it will return an empty string.

I find booleans are often easier to think about when they're the positive case - with `passwordsDoNotMatch` it does have immediate use in the thing you're writing right now, but in the future if we want to do something if they match we could find ourselves writing `!passwordsDoNotMatch`. `!passwordsMatch` and `passwordsMatch` are easier to use and understand.